bmc/routes.py

```diff
@@ -50,22 +50,33 @@ def enter():
 @login_required
 def save_voice():
     file = request.files['audio_data']
-    file.save(file.filename + ".wav")
+    file.save("bmc/recordings/" + book + "_" + chapter_num + ".wav")
     return "success" # "Done" button in page will move to new page
 
 @bmc_app.route('/bmc_trial')
 def memory():
     global book
     global library
-    global format
     library = Library()
-    format = "text"
-    try:
-        book = request.args.get("book")
-        return render_template('bmc_form.html', length = len(library.get(book)))
-    except TypeError:
+    global format
+    book = request.args.get("book")
+    format = request.args.get("format")
+    if (format == "text"):
+        try:
+            return render_template('bmc_form.html', length = len(library.get(book)))
+        except TypeError:
+            return render_template('error.html')
+    elif (format == "voice"):
+        return render_template('bmc_questions.html', length = len(library.get(book)))
```

Comment:
Since this line is outside of `try/except`, if someone writes the `book` name in the url, it'll trigger an exception that won't be caught. The `try/except` part should probably cover the whole `if/elif/else`:

```
    try:
        if (format == "text"):
            return render_template('bmc_form.html', length = len(library.get(book)))
        elif (format == "voice"):
            return render_template('bmc_questions.html', length = len(library.get(book)))
        else:
            return render_template('error.html')
    except:
        return render_template('error.html')
```